xbmc/cores/VideoPlayer/DVDCodecs/Video/MMALCodec.h

```diff
@@ -80,7 +80,7 @@ class CMMALVideo : public CDVDVideoCodec
   virtual void Reset(void);
   virtual bool GetPicture(DVDVideoPicture *pDvdVideoPicture);
   virtual bool ClearPicture(DVDVideoPicture* pDvdVideoPicture);
-  virtual unsigned GetAllowedReferences() { return 3; }
+  virtual unsigned GetAllowedReferences() { return NUM_BUFFERS; }
```

Comment:
We know 3 is too few, so needs to be at least 4. Any downside to having 6 (apart from an extra 2 video frames of gpu memory = ~6MB)?
I was assuming the extra frames would give more tolerance to a delay from decoder, for example.